cli/types/cypress.d.ts

```diff
@@ -2808,12 +2808,38 @@ declare namespace Cypress {
      */
     supportFile: string | false
     /**
-     * The test isolation level applied to ensure a clean slate between tests.
-     *   - legacy - resets/clears aliases, intercepts, clock, viewport, cookies, and local storage before each test.
-     *   - strict - applies all resets/clears from legacy, plus clears the page by visiting 'about:blank' to ensure clean app state before each test.
-     * @default "legacy", however, when experimentalSessionAndOrigin=true, the default is "strict"
-     */
-    testIsolation: 'legacy' | 'strict'
+     * The test isolation ensures a clean browser context between tests.
+     *
+     * Cypress will always resets/clears aliases, intercepts, clock, and viewport before each test
+     * to ensure a clean test slate; i.e. this configuration only impacts the browser context.
+     *
+     * Note: the `cy.session()` command will inherent this value to determine whether or not the page is
+     * cleared when the command executes.
+     *
+     * Options:
+     *   When experimentalSessionAndOrigin=false
+     *   - legacy - "Pass through" to document the default Cypress test isolation behavior. The page is not cleared
+     *          in end-to-end tests and cookies, local storage and session storage in the current domains are
+     *          cleared before each test. The `cy.session()` command is not available.
+     *          NOTE: this behavior will be changed in the next major release and the new default will be 'on'.
+     *
+     *   When experimentalSessionAndOrigin=true
+     *   - on - The page is cleared and cookies, local storage and session storage in all domains are cleared before each
+     *          test. The `cy.session()` command will also clear the page and current browser context when establishing
+     *          the browser session.
+     *   - off - The current browser state will persist between tests. The page does not clear and cookies, local storage and
+     *           session storage will be available in the next test. The `cy.session()` command will only clear the current
+     *           browser context when establishing the browser session - the current page will not clear.
+     *
+     * NOTES:
+     *    - Turning test isolation off may improve performance of end-to-end tests, however, previous tests could impact the
+     *      browser state of the next test and cause inconsistency when using .only().
```

Comment:
This could also have an impact if the user changes the order of their tests. I think what we are trying to convey is that tests are not guaranteed idempotency when run by themselves and I am not sure that is obvious.

Reply:
> This could also have an impact if the user changes the order of their tests.

This would depend on how the user has written their E2E tests -- you could write basic, repeatable tests that don't impact the state of the next test and could be ran standalone. Though most people may accidentally write tests that do rely on one another. If you / others feel strongly, I'd prefer to elaborate more in about the risks in the documentation vs the Typescript description....It might be worth removing this entire blur and adding an onlink to learn more info.

Reply:
I think that is pretty clear [here](https://github.com/cypress-io/cypress-documentation/pull/4797/files#diff-e8c9af19adcec43f11465b5acf72d7cd399b34f8015424b26bfaf9e482624434R657) with what you added. I wonder if we want to generalize this more as to your point, this is a TypeScript declaration file and not docs 😄 . Maybe something like:

> Turning test isolation off may improve performance of end-to-end tests, however, previous tests may not be idempotent. @see <LINK_TO_DOCS> 

which would give the clear description of what is meant here when we mention idempotent, since its probably an appropriate word to capture what we mean here, but not suitable for an in depth explanation which the docs provide.